lib/literal/types/range_type.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# @api private
+class RangeType < Literal::Type
+	def initialize(type)
+		@type = type
+	end
+
+	def inspect = "_Range(#{@type.inspect})"
+
+	def ===(value)
+		Range === value && (
+			(
+				@type === value.begin && (nil === value.end || @type === value.end)
+			) || (
+				@type === value.end && (nil === value.begin || @type === value.begin)
```

Comment:
Covers a range that ends with the expected type and is beginning-less. We've already covered the case where the range begins and ends with the expected type.